controllers/group_totals.go

```diff
@@ -56,48 +54,132 @@ func InstaCountHandler(rw http.ResponseWriter, r *http.Request) {
-	case "patient":
-		pipeline = models.NewPipeline(group)
-	case "encounter":
-		pipeline = models.NewEncounterPipeline(group)
-	case "condition":
-		pipeline = models.NewConditionPipeline(group)
-	}
-	qr, err := pipeline.ExecuteCount(server.Database)
-	if err != nil && err != mgo.ErrNotFound {
-		http.Error(rw, err.Error(), http.StatusInternalServerError)
-		return
+	pquery := ""
+	cquery := ""
+	equery := ""
+
+	intersectRequired := false
+
+	for _, characteristic := range group.Characteristic {
+
+		codings := characteristic.Code.Coding
+
+		for _, coding := range codings {
+			//Age
+			if coding.System == "http://loinc.org" && coding.Code == "21612-7" {
+				highAgeDate := time.Date(time.Now().Year()-int(*characteristic.ValueRange.Low.Value), time.Now().Month(), time.Now().Day(), 0, 0, 0, 0, time.UTC).Format("2006-01-02")
+				lowAgeDate := time.Date(time.Now().Year()-int(*characteristic.ValueRange.High.Value), time.Now().Month(), time.Now().Day(), 0, 0, 0, 0, time.UTC).Format("2006-01-02")
+				pquery += "birthdate=lt" + highAgeDate + "&birthdate=gt" + lowAgeDate + "&"
+				cquery += "patient.birthdate=lt" + highAgeDate + "&patient.birthdate=gt" + lowAgeDate + "&"
+				equery += "patient.birthdate=lt" + highAgeDate + "&patient.birthdate=gt" + lowAgeDate + "&"
+			}
+
+			//Gender
+			if coding.System == "http://loinc.org" && coding.Code == "21840-4" {
+				gender := characteristic.ValueCodeableConcept.Coding[0].Code
+				pquery += "gender=" + gender + "&"
+				cquery += "patient.gender=" + gender + "&"
+				equery += "patient.gender=" + gender + "&"
+			}
+
+			//Condition
+			if coding.System == "http://loinc.org" && coding.Code == "11450-4" {
+				cquery += "code=" + characteristic.ValueCodeableConcept.Coding[0].System + "|" + characteristic.ValueCodeableConcept.Coding[0].Code + "&"
+				intersectRequired = true
+			}
+
+			//Encounter
+			if coding.System == "http://loinc.org" && coding.Code == "46240-8" {
+				equery += "type=" + characteristic.ValueCodeableConcept.Coding[0].System + "|" + characteristic.ValueCodeableConcept.Coding[0].Code + "&"
+				intersectRequired = true
+			}
+		}
 	}
-	json.NewEncoder(rw).Encode(qr)
-}
 
-func InstaCountAllHandler(rw http.ResponseWriter, r *http.Request) {
-	decoder := json.NewDecoder(r.Body)
-	group := &fhirmodels.Group{}
-	err := decoder.Decode(group)
-	if err != nil {
-		http.Error(rw, err.Error(), http.StatusInternalServerError)
+	pquery = TrimSuffix(pquery, "&")
+	cquery = TrimSuffix(cquery, "&")
+	equery = TrimSuffix(equery, "&")
+
+	searcher := search.NewMongoSearcher(server.Database)
+
+	var pResultIDs []struct {
+		ID string `bson:"_id"`
+	}
+	pSearchQuery := search.Query{Resource: "Patient", Query: pquery}
+	pQ := searcher.CreateQuery(pSearchQuery)
+	pQ.Select(bson.M{"_id": 1}).All(&pResultIDs)
+	pids := make([]string, len(pResultIDs))
+	for i := range pResultIDs {
+		pids[i] = pResultIDs[i].ID
 	}
 
-	pipelineMap := make(map[string]models.Pipeline)
+	if intersectRequired {
```

Comment:
I think it would be good to add some comments to briefly summarize what `intersectRequired` means (when it's needed, why it's needed, what it does).  For someone reading the code the first time, they might have trouble deciphering what's going on.